src/databricks/labs/remorph/install.py

```diff
@@ -0,0 +1,202 @@
+        logger.info("Please answer a couple of questions to configure Remorph")
+
+        source_prompt = self._prompts.choice("Select the source", ["snowflake", "tsql"])
+        source = source_prompt.lower()
+
+        skip_validation = self._prompts.confirm("Do you want to Skip Validation")
+
+        catalog_name = self._prompts.question("Enter catalog_name")
+
+        try:
+            self._catalog_setup.get(catalog_name)
+        except NotFound:
+            self.setup_catalog(catalog_name)
+
+        schema_name = self._prompts.question("Enter schema_name")
+
+        try:
+            self._catalog_setup.get_schema(f"{catalog_name}.{schema_name}")
+        except NotFound:
+            self.setup_schema(catalog_name, schema_name)
+
+        config = MorphConfig(
+            source=source,
+            skip_validation=skip_validation,
+            catalog_name=catalog_name,
+            schema_name=schema_name,
+            sdk_config=None,
+        )
+
+        ws_file_url = self._installation.save(config)
+        if self._prompts.confirm("Open config file in the browser and continue installing?"):
+            webbrowser.open(ws_file_url)
+        return config
+
+    @retried(on=[NotFound], timeout=timedelta(minutes=5))
+    def setup_catalog(self, catalog_name: str):
+        allow_catalog_creation = self._prompts.question(
+            f"""Catalog {catalog_name} not found.\
+                    \nDo you want to create a new one?"""
+        )
+        if not allow_catalog_creation:
+            msg = "Catalog is needed to setup Remorph"
+            raise SystemExit(msg)
+        else:
+            logger.info(f" Creating new Catalog {catalog_name}")
+            self._catalog_setup.create(catalog_name)
+
+    @retried(on=[NotFound], timeout=timedelta(minutes=5))
+    def setup_schema(self, catalog_name: str, schema_name: str):
+        allow_schema_creation = self._prompts.question(
+            f"""Schema {schema_name} not found in Catalog {catalog_name}\
+                    \nDo you want to create a new Schema?"""
+        )
+        if not allow_schema_creation:
+            msg = "Schema is needed to setup Remorph"
+            raise SystemExit(msg)
+        else:
+            logger.info(f" Creating new Schema {catalog_name}.{schema_name}")
+            self._catalog_setup.create_schema(schema_name, catalog_name)
+
+
+class WorkspaceInstallation:
+    def __init__(
+        self,
+        config: MorphConfig,
+        installation: Installation,
+        ws: WorkspaceClient,
+        prompts: Prompts,
+        verify_timeout: timedelta,
+    ):
+        self._config = config
+        self._installation = installation
+        self._ws = ws
+        self._prompts = prompts
+        self._verify_timeout = verify_timeout
+        self._state = InstallState.from_installation(installation)
+        self._this_file = Path(__file__)
+
+    @classmethod
+    def current(cls, ws: WorkspaceClient):
```

Comment:
You don't need this method. Using Installation.current(ws) through the code is enough.

Besides - other code should not depend on installer script, otherwise we mess up the sys.path

Reply:
Done